### PR TITLE
Add intent tab

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Goonj Authorize fields came under Intent form

image

### CODE REVIEW

**Review by pokhiii — COMMENTED** [review]
left some comments

File: wp-content/civi-extensions/goonjcustom/goonjcustom.php [hunk 1]
```diff
@@ -277,4 +300,43 @@ function goonjcustom_civicrm_tabset($tabsetName, &$tabs, $context) {
 		'active' => 1,
 		'current' => false,
 	];
+
+	// Add JavaScript to dynamically hide fields
+	CRM_Core_Region::instance('page-footer')->add(
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/goonjcustom.php:305]: Is there a check we can add to load this JS only the page where we need it?

File: wp-content/civi-extensions/goonjcustom/goonjcustom.php [hunk 2]
```diff
@@ -254,6 +255,19 @@ function goonjcustom_civicrm_tabset($tabsetName, &$tabs, $context) {
...
+	->addSelect('Event_Volunteers.Collection_Camp_Intent')
+	->addWhere('id', '=', $eventID)
+	->setLimit(1)
+	->execute();
+
+	$collectionCampIntentId= $intentId->first()['Event_Volunteers.Collection_Camp_Intent'] ?? null;
+
+	// URL for the Intent tab
+	$intenturl = CRM_Utils_System::url(
+		"/wp-admin/admin.php?page=CiviCRM&q=civicrm%2Factivity%2Fadd&reset=1&type=Activity&subType=61&action=view&id=$collectionCampIntentId"
```
[the remark this answers is not in the document]
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/goonjcustom.php]: I'm hardcoding it to hide this item based on the condition below, but I think that's not a good approach. Now I'm taking the collectionCampEventTypeId dynamically and hides. the field on based on this.

File: wp-content/civi-extensions/goonjcustom/goonjcustom.php (hunk 1 above)
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/goonjcustom.php]: yes

**Review by pokhiii — APPROVED** [review]

**Pull request merged**